File: src/renders/habitats/LandHabitat.cpp
/* NIM/Nama  : 13515AAA / AAA       */
/* File      : Habitat.cpp              */

#include "LandHabitat.h"

using namespace std;

LandHabitat::LandHabitat() : Habitat('X', WHITE)
{
}
LandHabitat::LandHabitat(const LandHabitat& L) : Habitat(L)
{
}
LandHabitat::~LandHabitat()
{
}
LandHabitat& LandHabitat::operator=(const LandHabitat& L)
{
	Habitat::operator=(L);
	return *this;
}
LandHabitat* LandHabitat::clone() const
{
	return new LandHabitat(*this);
}
char LandHabitat::render()
{
	return id;
}
Color LandHabitat::getColor()
{
	return color;
}

File: src/habitat/land/LandHabitat.java
/**
 * @author NIM/Nama : 13515078 / Veren Iliana Kurniadi
 * File : LandHabitat.java
 */

package habitat.land;

import java.util.*;

import habitat.Habitat;

import java.lang.*;

public class LandHabitat extends Habitat{
   /**
   * Constructor
   */
  public LandHabitat(){
    super();
  }
  /**
   * Mengembalikan karakter untuk ditampilkan ke layar
   * @return id bertipe char
   */
  public char render(){
    return 'L';
  }
  /**
   * Mengembalikan warna untuk ditampilkan ke layar
   * @return color warna renderable
   */
  public String renderWithColor(){
    return ANSI_WHITE + render() + ANSI_RESET;
  }
}
